signer/rsapss/rsapss.go

```diff
@@ -0,0 +1,170 @@
+package rsapss
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/ThalesIgnite/crypto11"
+	"github.com/pkg/errors"
+	"go.mozilla.org/autograph/signer"
+)
+
+const (
+	// Type of this signer is "rsapss"
+	Type = "rsapss"
+)
+
+// RSAPSSSigner holds the configuration of the signer
+type RSAPSSSigner struct {
+	signer.Configuration
+
+	// key is the RSA private key to sign hashes
+	key crypto.PrivateKey
+
+	// rng is our random number generator
+	rng io.Reader
+}
+
+// New initializes a pgp signer using a configuration
+func New(conf signer.Configuration) (s *RSAPSSSigner, err error) {
+	s = new(RSAPSSSigner)
+
+	if conf.Type != Type {
+		return nil, errors.Errorf("rsapss: invalid type %q, must be %q", conf.Type, Type)
+	}
+	s.Type = conf.Type
+
+	if conf.ID == "" {
+		return nil, errors.New("rsapss: missing signer ID in signer configuration")
+	}
+	s.ID = conf.ID
+
+	if conf.PrivateKey == "" {
+		return nil, errors.New("rsapss: missing private key in signer configuration")
+	}
+	s.PrivateKey = conf.PrivateKey
+
+	if conf.PublicKey == "" {
+		return nil, errors.New("rsapss: missing public key in signer configuration")
+	}
+	s.key, _, s.rng, s.PublicKey, err = conf.GetKeysAndRand()
+	if err != nil {
+		return nil, errors.Wrapf(err, "rsapss: error fetching key and rand from signer configuration")
+	}
+
+	switch s.key.(type) {
+	case *rsa.PrivateKey, *crypto11.PKCS11PrivateKeyRSA:
```

Comment:
nit: crypto11 isn't really needed here. Signers can simply rely on the crypto.Signer interface those keys implement, and you can actually rewrite this as follows:
```go
switch s.pub.(type) {
case *rsa.PublicKey:
```